src/slic3r/GUI/GUI_ObjectList.cpp

```diff
@@ -304,6 +310,7 @@ ObjectList::ObjectList(wxWindow* parent) :
 
 ObjectList::~ObjectList()
 {
+    delete m_objects_model;
```

Comment:
This line is causing crash when closing the application. Hence probably why the original BBL code did not have the destructor invoked.